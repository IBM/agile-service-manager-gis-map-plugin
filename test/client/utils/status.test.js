// Import the module under test
import * as status from '../../../src/scripts/client/utils/status';

describe('Status', function() {
    test('severityRank defaults', function() {
        expect(status.severityRank).toMatchSnapshot();
    });
    test('severityColors defaults', function() {
        expect(status.severityColors).toMatchSnapshot();
    });

    test('propertySortHighToLow lower value with rank map', function() {
        const sort = status.propertySortHighToLow('test', status.severityRank)
        expect(sort({test: 'clear'}, {test: 'major'})).toEqual(1);
    });

    test('propertySortHighToLow same value with rank map', function() {
        const sort = status.propertySortHighToLow('test', status.severityRank)
        expect(sort({test: 'clear'}, {test: 'clear'})).toEqual(0);
    });

    test('propertySortHighToLow higher value with rank map', function() {
        const sort = status.propertySortHighToLow('test', status.severityRank)
        expect(sort({test: 'warning'}, {test: 'clear'})).toEqual(-1);
    });

    test('propertySortHighToLow lower value without rank map', function() {
        const sort = status.propertySortHighToLow('test')
        expect(sort({test: 'a'}, {test: 'f'})).toEqual(1);
    });

    test('propertySortHighToLow same value without rank map', function() {
        const sort = status.propertySortHighToLow('test')
        expect(sort({test: 'b'}, {test: 'b'})).toEqual(0);
    });

    test('propertySortHighToLow higher value without rank map', function() {
        const sort = status.propertySortHighToLow('test')
        expect(sort({test: 'z'}, {test: 'b'})).toEqual(-1);
    });


    test('propertySortLowToHigh lower value with rank map', function() {
        const sort = status.propertySortLowToHigh('test', status.severityRank)
        expect(sort({test: 'clear'}, {test: 'major'})).toEqual(-1);
    });

    test('propertySortLowToHigh same value with rank map', function() {
        const sort = status.propertySortLowToHigh('test', status.severityRank)
        expect(sort({test: 'clear'}, {test: 'clear'})).toEqual(0);
    });

    test('propertySortLowToHigh higher value with rank map', function() {
        const sort = status.propertySortLowToHigh('test', status.severityRank)
        expect(sort({test: 'warning'}, {test: 'clear'})).toEqual(1);
    });

    test('propertySortLowToHigh lower value without rank map', function() {
        const sort = status.propertySortLowToHigh('test')
        expect(sort({test: 'a'}, {test: 'f'})).toEqual(-1);
    });

    test('propertySortLowToHigh same value without rank map', function() {
        const sort = status.propertySortLowToHigh('test')
        expect(sort({test: 'b'}, {test: 'b'})).toEqual(0);
    });

    test('propertySortLowToHigh higher value without rank map', function() {
        const sort = status.propertySortLowToHigh('test')
        expect(sort({test: 'z'}, {test: 'b'})).toEqual(1);
    });

    
});