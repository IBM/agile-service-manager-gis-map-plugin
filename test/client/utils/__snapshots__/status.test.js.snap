// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Status severityColors defaults 1`] = `
Object {
  "critical": "#da1e28",
  "indeterminate": "#4F2196",
  "information": "#054ADA",
  "major": "#fc991e",
  "minor": "#fdd13a",
  "warning": "#408BFC",
}
`;

exports[`Status severityRank defaults 1`] = `
Object {
  "clear": 1,
  "critical": 7,
  "indeterminate": 2,
  "information": 3,
  "major": 6,
  "minor": 5,
  "unknown": 0,
  "warning": 4,
}
`;
